src/modules/musig/musig.md

```diff
@@ -0,0 +1,189 @@
+       their partial signatures. They do so by calling `secp256k1_musig_session_combine_nonces`
+       which updates in place
+           * the session state `session`
+           * the array of signer data `signers`
+       It outputs an auxillary integer `nonce_is_negated` and has an auxillary input
+       `adaptor`. Both of these may be set to NULL for ordinary signing purposes.
+       If the signer did not provide a message to `secp256k1_musig_session_initialize`,
+       a message must be provided now by calling `secp256k1_musig_session_set_msg` which
+       updates the session state in place.
+    6. The signer computes a partial signature `s_i` using the function
+       `secp256k1_musig_partial_sign` which takes the session state as input and
+       partial signature as output.
+    7. The signer then communicates the partial signature `s_i` to all other signers, or
+       to a central coordinator. These partial signatures should be serialized using
+       `musig_partial_signature_serialize` and parsed using `musig_partial_signature_parse`.
+    8. Each signer calls `secp256k1_musig_partial_sig_verify` on the other signers' partial
+       signatures to verify their correctness. If only the validity of the final signature
+       is important, not assigning blame, this step can be skipped.
+    9. Any signer, or central coordinator, may combine the partial signatures to obtain
+       a complete signature using `secp256k1_musig_partial_sig_combine`. This function takes
+       a signing session and array of MuSig partial signatures, and outputs a single
+       Schnorr signature.
+
+### Non-signing Participant
+
+A participant who wants to verify the signing process but not actually contribute a
+partial signature, may do so using the above instructions except for the following
+changes:
+
+    1. A signing session should be produced using `musig_session_initialize_verifier`
+       rather than `musig_session_initialize`; this function takes no secret data or
+       signer index.
+    2. The participant receives nonce commitments, public nonces and partial signatures,
+       but does not produce these values. Therefore `secp256k1_musig_session_get_public_nonce`
+       and `secp256k1_musig_partial_sign` are not called.
+
+### Verifier
+
+The final signature is simply a valid Schnorr signature using the combined public key. It
+can be verified using the `secp256k1_schnorrsig_verify` with the correct message and
+public key output from `secp256k1_musig_pubkey_combine`.
+
+## Atomic Swaps
+
+The signing API supports the production of "adaptor signatures", modified partial signatures
+which have an auxiliary secret encrypted to them. A partial signature is revealed to a party
+who learns the secret; conversely, a party with the corresponding secret will learn the
+secret.
+
+Using these adaptor signatures, two 2-of-2 MuSig signing protocols can be executed in
```

Comment:
Maybe before explaining atomic swap in detail we could more abstractly explain what the adaptor signature is. Something like
```
Given a signing session where combine_nonces was called with public key `T = t*G` as `adaptor`, and partial signatures from all participants (verified)
    1. This session's combined signature must be made by one party adapting their partial signature with `t` and combining it with the remaining partial signatures.
    2. All participants can compute `t` from the session's combined signature and all previously received (adaptorless/un-adapted?) partial signatures
```

Reply:
Reworded.